app/react-components/components/SurvivalPlotWrapper.js

```diff
@@ -91,59 +86,104 @@ class SurvivalPlotWrapper extends Component {
   constructor(props) {
     super(props);
 
-    this.onDomainChange = this.onDomainChange.bind(this);
-
     this.state = {
       xDomain: undefined,
       stack: [],
       palette: [colors(0), colors(1)],
+      disabledDataSets: undefined,
+      dataSets: null,
     };
+  }
 
+  componentDidMount() {
     buildData(this.props).then(d => this.setState(d));
   }
 
+  componentWillReceiveProps(nextProps) {
+    buildData(nextProps).then((newState) => {
+      if (!_.isEqual(newState, this.state)) {
+        this.setState({
+          ...this.resetXDomain(),
+          ...newState,
+        });
+      }
+    });
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    return !_.isEqual(this.props, nextProps) || !_.isEqual(this.state, nextState);
+  }
+
+  componentDidUpdate() {
+    this.update();
+  }
+
   onDomainChange(xDomain) {
     this.setState({
       stack: this.state.stack.concat(this.state.xDomain),
       xDomain,
     });
   }
 
-  resetXDomain() {
-    if(this.state.stack.length){
-      return {
-        stack: [],
-        xDomain: this.state.stack[0],
-      };
-    } else {
-      return {};
+  update() {
+    const { disabledDataSets, dataSets, palette, xDomain } = this.state;
+    const {
+      height = 0,
+      getSetSymbol,
+      margins = {
+        top: 15,
+        right: 20,
+        bottom: 40,
+        left: 50,
+      },
+    } = this.props;
+
+    const container = this.survivalContainer;
+    const onDomainChange = this.onDomainChange.bind(this);
+
+    if (dataSets) {
+      renderPlot({
+        container,
+        dataSets,
+        disabledDataSets,
+        palette,
+        xDomain,
+        xAxisLabel: 'Duration (days)',
+        yAxisLabel: 'Survival Rate',
+        height: isFullScreen(container) ? (window.innerHeight - 100) : height,
+        getSetSymbol,
+        onMouseEnterDonor: () => {},
+        onMouseLeaveDonor: () => {},
+        onClickDonor: () => {},
+        onDomainChange,
+        margins,
+      });
     }
   }
 
-  reset() {
-    this.props.onReset();
+  resetXDomain() {
+    const { stack, xDomain } = this.state;
+
+    return {
+      stack: [],
+      xDomain: stack.length ? stack[0] : xDomain,
+    };
   }
 
-  componentWillReceiveProps(nextProps) {
-    buildData(nextProps).then(newState => {
-      this.setState({
-        ...this.resetXDomain(),
-        ...newState,
-      });
-    });
+  reset() {
+    this.props.onReset();
   }
 
   render() {
-    const { xDomain, data, legend, pValue, palette } = this.state;
-    const { height, width } = this.props;
+    const { legend, pValue, palette } = this.state;
 
     return (
       <div
         className="survival-plot"
-        ref={el => this.container = el}
+        ref={(el) => { this.container = el; }}
```

Comment:
It's used to get the SVG for downloading, but theres actually another ref that makes this one redundant, I'll remove this one.